brayns/common/scene/Model.cpp

```diff
@@ -118,10 +118,61 @@ void Model::addModel(ModelPtr model, const Transformations& transform)
     _modelsDirty = true;
 }
 
+uint64_t Model::addSDFGeometry(const size_t materialId, const SDFGeometry& geom,
+                               const std::vector<size_t>& neighbourIndices)
+{
+    _sdfGeometryData.isDirty = true;
+    const uint32_t geomIdx = _sdfGeometryData.geometries.size();
+    _sdfGeometryData.geometryIndices[materialId].push_back(geomIdx);
+
+    assert(_sdfGeometryData.geometries.size() ==
+           _sdfGeometryData.neighbours.size());
+
+    _sdfGeometryData.neighbours.push_back(neighbourIndices);
+    _sdfGeometryData.geometries.push_back(geom);
+
+    switch (geom.type)
+    {
+    case brayns::SDFType::Sphere:
+    {
+        _bounds.merge(geom.center - Vector3f(geom.radius));
+        _bounds.merge(geom.center + Vector3f(geom.radius));
+        break;
+    }
+    case brayns::SDFType::Pill:
+    case brayns::SDFType::ConePill:
+    case brayns::SDFType::ConePillSigmoid:
+    {
+        const auto min = Vector3f{
+            std::min(geom.p0.x() - geom.radius, geom.p1.x() - geom.radius),
+            std::min(geom.p0.y() - geom.radius, geom.p1.y() - geom.radius),
+            std::min(geom.p0.z() - geom.radius, geom.p1.z() - geom.radius)};
+        const auto max = Vector3f{
+            std::max(geom.p0.x() + geom.radius, geom.p1.x() + geom.radius),
+            std::max(geom.p0.y() + geom.radius, geom.p1.y() + geom.radius),
+            std::max(geom.p0.z() + geom.radius, geom.p1.z() + geom.radius)};
+
+        _bounds.merge(min);
+        _bounds.merge(max);
+        break;
+    }
+    default:
+        throw std::runtime_error("No bounds found for SDF type.");
+    }
```

Comment:
I would refactor this a bit, moving the logic to get the min/max to the geometry class and keep here only something like:
_bounds.merge(geom.getMin());
_bounds.merge(geom.getMax());
As an added benefit, it lets you unit test independently the geom.getMin/Max() code :-)

Reply:
Yes, I created a function that returns the bounding box.